pkg/disk/attachdetach_slot.go

```diff
@@ -0,0 +1,80 @@
+package disk
+
+import (
+	"context"
+	"sync"
+)
+
+type AttachDetachSlots struct {
+	nodes       sync.Map
+	slotFactory func() slot
+}
+
+func (a *AttachDetachSlots) GetSlotFor(node string) slot {
+	value, _ := a.nodes.LoadOrStore(node, a.slotFactory())
+	return value.(slot)
+}
+
+type slot interface {
+	AquireDetach(ctx context.Context) error
+	AquireAttach(ctx context.Context) error
+	Release()
+}
+
+type serialSlot struct {
+	detachSlot chan struct{}
```

Comment:
slot & priority slot would be better

Reply:
OK, I've updated the naming, and added some comments.